### PR TITLE
Add CONTRIBUTING.md for scratch-vm

### PR DESCRIPTION
**Pull request by tmickel**
Proposed changes

Adds contributor guidelines for the repository.

Reason for changes

To provide some context to contributors about the project and productive guidelines for contributions :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Two great ways of helping are by identifying bugs and documenting them as issues,
+or fixing issues and creating pull requests. When submitting pull requests please be patient
+-- it can take a while to find time to review them.
+The organization and class structures can't be radically changed without significant coordination
+and collaboration from the Scratch Team, so these types of changes should be avoided.
+
+It's been said that the Scratch Team spends about one hour of design discussion for every pixel in Scratch,
+but some think that estimate is a little low. While we welcome suggestions for new features in our
+[suggestions forum](https://scratch.mit.edu/discuss/1/) (especially ones that come with mockups), we are unlikely to accept PRs with
+new features that we haven't deeply thought through. Why? Because we have a strong belief
```
Comment (thisandagain) [inline CONTRIBUTING.md]: I'm wondering if we might want to change "deeply thought through" with "been thought through and discussed as a group".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+in the value of keeping things simple for new users. To learn more about our design philosophy,
+see [this forum post](http://scratch.mit.edu/discuss/post/1576/), or
```
Comment (thisandagain) [inline CONTRIBUTING.md]: I'd rather direct folks to:
https://scratch.mit.edu/developers

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+[this paper](http://web.media.mit.edu/~jmaloney/papers/ScratchLangAndEnvironment.pdf).
```
Comment (thisandagain) [inline CONTRIBUTING.md]: Perhaps this paper would be a better fit (which discusses these values more broadly):
http://web.media.mit.edu/~silver/radlearning/construction_kits_for_kids.pdf

... or alternatively this paper from 2009 which still is specific to Scratch but talks about the goals a bit more eloquently:
http://web.media.mit.edu/~mres/papers/Scratch-CACM-final.pdf

**Comment by thisandagain** [comment]
Thanks for putting this together. A few small comments but otherwise this is great.

**Comment by tmickel** [comment]
Ok, updated. Any more thoughts? Note that we might want to update the scratch-flash readme, as most of the changes were copied directly from there.

**Comment by thisandagain** [comment]
Looks good. Thanks for the heads-up on the `scratch-flash` README. I didn't realize that was the origin point for this.

**Pull request merged**